sceptre/template_handlers/helper.py

```diff
@@ -142,5 +149,17 @@ def render_jinja_template(path, jinja_vars, j2_environment):
 
     template = j2_environment.get_template(path.name)
 
-    body = template.render(**jinja_vars)
+    try:
+        body = template.render(**jinja_vars)
+    except Exception as err:
+        message = f"{path} - {err}"
+
+        if logging_level() == logging.DEBUG:
+            debug_file_path = write_debug_file(
+                json.dumps(jinja_vars, indent=4), prefix="vars_"
+            )
+            message += f"\nTemplating vars saved to: {debug_file_path}"
```

Comment:
errors from this method are usually handled by `print_template_traceback`, would it make sense to put this stuff in `print_template_traceback` instead?

Reply:
Condensed version of discussion Slack:
1. The `print_template_traceback` function prints a formatted stack trace; this code is writing out the templating vars (and in other contexts the rendered template) to a file for debugging purposes.
2. There is duplication of Jinja rendering in this code base, with separate Jinja rendering implemented in the config reader and templating classes. Meanwhile this debug file is used in both but primarily in the reader class, where the `print_template_traceback` is not available. It could of course be moved, but point (1) would need to be addressed first.

Reply:
ok, I have taken a closer look and you have convinced me.  thanks.